client/jetpack-connect/index.js

```diff
@@ -44,15 +44,16 @@ export default function() {
 		page.redirect( `/jetpack/connect/store${ params.interval ? '/' + params.interval : '' }` )
 	);
 
+	page( '/jetpack/connect/plans', '/jetpack/connect/store' );
```

Comment:
We could add this to the preceeding rule, it keeps an optional `:interval` as well:

```js
page( '/jetpack/connect/:_(plans|akismet|vaultpress)/:interval(yearly|monthly)?', ( { params } ) =>
  page.redirect( `/jetpack/connect/store${ params.interval ? '/' + params.interval : '' }` )
);
```